File: Voting-System-main/src/pages/candidate.js
import axios from 'axios';
import React from 'react';
import { useState } from "react";
import { useHistory } from 'react-router-dom';
import "../components/registration.css";
function Candidate() {
    let history = useHistory();
    const [name, setname] = useState("");
    const [email, setemail] = useState("");
    const [designation, setdesignation] = useState("");
    const [bio, setbio] = useState("");
    const [electionid, setelectionid] = useState("");

    async function handleSubmit(event) {
        event.preventDefault();
        try {

            await axios.post("http://localhost:8004/saveCandidate",
                {

                    name: name,
                    email: email,
                    designation: designation,
                    bio: bio,
                    electionid: electionid,
                    counts: 0

                });

            alert("User Registation Successfully");
            setname("");
            setemail("");
            setdesignation("");
            setbio("");
            setelectionid("");




        }
        catch (err) {
            alert("User Registation Failed" + err);
        }
    }


    return (
        <div className='bb'>
            <br></br>
            <br></br>
            {window.candidatelist != "" ? <div className="register-container" >
                <div class="split left">
                    <div class="centered">

                        <h1>Candidates List</h1>
                        {window.candidatelist.map((e) => {
                            const list = (
                                <>
                                    <ul>
                                        <div className='card'>

                                            <p>{e.name}</p>
                                            {/* <p>{e.email}</p>
                                            <p>{e.designation}</p> */}
                                        </div>
                                        <br></br>

                                    </ul>
                                    <hr />
                                </>
                            );
                            return list;
                        })}
                    </div>
                </div>

                <div class="split right">
                    <div class="centered">


                        <form className="register-form" onSubmit={handleSubmit}>
                            <br></br>
                            <h1>Create Candidate</h1>
                            <p>Fill in the Information Below</p>

                            <input type="text"
                                name="name"
                                placeholder="name"
                                size={60}
                                onChange={(event) => {
                                    setname(event.target.value);
                                }}
                            />

                            <input type="text"
                                name="email"
                                placeholder="email"
                                onChange={(event) => {
                                    setemail(event.target.value);
                                }}
                            />

                            <input type="text"
                                name="designation"
                                placeholder="designation"

                                onChange={(event) => {
                                    setdesignation(event.target.value);
                                }}
                            />
                            <input type="text"
                                name="bio"
                                placeholder="bio"

                                onChange={(event) => {
                                    setbio(event.target.value);
                                }}
                            />
                            <input type="text"
                                name="electionid"
                                placeholder="electionid"

                                onChange={(event) => {
                                    setelectionid(event.target.value);
                                }}
                            />






                            <button type="submit">Create</button>


                        </form>


                    </div>
                </div>

            </div >
                //second one
                : <div class="centered">


                    <form className="register-form" onSubmit={handleSubmit}>
                        <br></br>
                        <h1>Create Candidate</h1>
                        <p>Fill in the Information Below</p>

                        <input type="text"
                            name="name"
                            placeholder="name"
                            size={60}
                            onChange={(event) => {
                                setname(event.target.value);
                            }}
                        />

                        <input type="text"
                            name="email"
                            placeholder="email"
                            onChange={(event) => {
                                setemail(event.target.value);
                            }}
                        />

                        <input type="text"
                            name="designation"
                            placeholder="designation"

                            onChange={(event) => {
                                setdesignation(event.target.value);
                            }}
                        />
                        <input type="text"
                            name="bio"
                            placeholder="bio"

                            onChange={(event) => {
                                setbio(event.target.value);
                            }}
                        />
                        <input type="text"
                            name="electionid"
                            placeholder="electionid"

                            onChange={(event) => {
                                setelectionid(event.target.value);
                            }}
                        />






                        <button type="submit">Create</button>


                    </form>


                </div>}
            <br></br> <br></br>
        </div>
    );
};

export default Candidate;

File: candidates/src/main/java/com/example/candidates/entity/Candidate.java
package com.example.candidates.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="Candidates")
public class Candidate 
{	
   

	@Id
    @Column(name="id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
	
    @Column(name="name")
    private String name;

    
   
    
    @Column(name="email")
    private String email;
    
    @Column(name="designation")
    private String designation;
    
    @Column(name="bio")
    private String bio;
    
    @Column(name="electionid")
    private String electionid;
    
    @Column(name="counts")
    private int counts;
 
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getDesignation() {
		return designation;
	}

	public void setDesignation(String designation) {
		this.designation = designation;
	}

	public String getBio() {
		return bio;
	}

	public void setBio(String bio) {
		this.bio = bio;
	}

	public String getElectionid() {
		return electionid;
	}

	public void setElectionid(String electionid) {
		this.electionid = electionid;
	}
	   
    public int getCounts() {
		return counts;
	}

	public void setCounts(int counts) {
		this.counts = counts;
	}

}
